doc/hash2/sha-2.adoc

```diff
@@ -0,0 +1,260 @@
+### <boost/hash2/sha2.hpp>
```

Comment:
Heading level should start from one, and be adjusted with leveloffset in the include. The .adoc should probably be sha2.adoc (like sha2.hpp), not sha-2.adoc.